[MM-61102] Fix errcheck issues in server/channels/app/channel_bookmark_test.go

RS-labhub opened the pull request:
#### Summary
Fixed the errcheck issues

#### Ticket Link
fixes #28782 
Jira [MM-61102](https://mattermost.atlassian.net/browse/MM-61102)

#### Release Note
```release-note
NONE
```

[MM-61102]: https://mattermost.atlassian.net/browse/MM-61102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -98,8 +98,13 @@ func TestUpdateBookmark(t *testing.T) {
 
-		th.App.Srv().Store().FileInfo().Save(th.Context, file)
-		defer th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file.Id)
+		if _, appErr := th.App.Srv().Store().FileInfo().Save(th.Context, file); appErr != nil {
+			assert.Nil(t, appErr)
+		}
```
We actually don't need the `if` checks.

RS-labhub replied:
That was my mistake... sorry for that.. I was shortening the code and forget to remove that. So sorry!

hanzei reviewed (changes requested):
There are still some linter errors. Can you please take a look? Thanks!

```
Error: channels/app/channel_bookmark_test.go:102:4: calling assert.Nil on error, please use assert.NoError instead
Error: channels/app/channel_bookmark_test.go:106:4: calling assert.Nil on error, please use assert.NoError instead
Error: channels/app/channel_bookmark_test.go:130:4: calling assert.Nil on error, please use assert.NoError instead
Error: channels/app/channel_bookmark_test.go:133:4: calling assert.Nil on error, please use assert.NoError instead
Error: channels/app/channel_bookmark_test.go:137:4: calling assert.Nil on error, please use assert.NoError instead
Error: channels/app/channel_bookmark_test.go:282:3: calling assert.Nil on error, please use assert.NoError instead
Error: channels/app/channel_bookmark_test.go:286:3: calling assert.Nil on error, please use assert.NoError instead
Error: channels/app/channel_bookmark_test.go:377:3: calling require.Nil on error, please use require.NoError instead
```

hanzei commented:
/update-branch

RS-labhub commented:
@hanzei how to check the linter tests?

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -102,8 +102,12 @@ func TestUpdateBookmark(t *testing.T) {
 
-		th.App.Srv().Store().FileInfo().Save(th.Context, file)
-		defer th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file.Id)
+		_, appErr := th.App.Srv().Store().FileInfo().Save(th.Context, file)
+		assert.NoError(t, appErr)
```
Nit: `Save` return an `error`, hence the variable name should be `err` instead of `appErr`

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -125,9 +129,14 @@ func TestUpdateBookmark(t *testing.T) {
-		th.App.Srv().Store().FileInfo().AttachToPost(th.Context, file2.Id, model.NewId(), th.BasicChannel.Id, model.BookmarkFileOwner)
-		defer th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file2.Id)
+		_, appErr = th.App.Srv().Store().FileInfo().Save(th.Context, file2)
+		assert.NoError(t, appErr)
```
Same here

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -125,9 +129,14 @@ func TestUpdateBookmark(t *testing.T) {
+		_, appErr = th.App.Srv().Store().FileInfo().Save(th.Context, file2)
+		assert.NoError(t, appErr)
+		appErr = th.App.Srv().Store().FileInfo().AttachToPost(th.Context, file2.Id, model.NewId(), th.BasicChannel.Id, model.BookmarkFileOwner)
+		assert.NoError(t, appErr)
```
And here

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -279,7 +293,9 @@ func TestGetChannelBookmarks(t *testing.T) {
 	}
 
-	th.App.CreateChannelBookmark(th.Context, bookmark2, "")
+	if _, appErr := th.App.CreateChannelBookmark(th.Context, bookmark2, ""); appErr != nil {
+		assert.Nil(t, appErr)
+	}
```
We don't need the `if` check

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -290,7 +306,9 @@ func TestGetChannelBookmarks(t *testing.T) {
 	t.Run("get bookmarks of a channel after one is deleted (aka only return the changed bookmarks)", func(t *testing.T) {
 		now := model.GetMillis()
-		th.App.DeleteChannelBookmark(bookmark1.Id, "")
+		if _, appErr := th.App.DeleteChannelBookmark(bookmark1.Id, ""); appErr != nil {
+			assert.Nil(t, appErr)
+		}
```
Same here

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -437,7 +458,9 @@ func TestUpdateChannelBookmarkSortOrder(t *testing.T) {
 
 		// now reset order
-		th.App.UpdateChannelBookmarkSortOrder(bookmark0.Id, channelId, int64(0), "")
+		if _, appErr = th.App.UpdateChannelBookmarkSortOrder(bookmark0.Id, channelId, int64(0), ""); appErr != nil {
+			assert.Nil(t, appErr)
+		}
```
And here

hanzei reviewed (changes requested):
CI is green now :rocket: 

Please take a look at my comments below

RS-labhub reviewed (commented):
Renamed and removed unnecessary lines

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -125,9 +129,14 @@ func TestUpdateBookmark(t *testing.T) {
+		}()
 
 		bookmark2.FileId = file2.Id
 		bookmarkResp, err = th.App.CreateChannelBookmark(th.Context, bookmark2, "")
```
This also needs to be renamed to `*model.AppError`.

RS-labhub replied:
For this too? https://github.com/mattermost/mattermost/pull/28789#discussion_r1816418394

hanzei replied:
Yes, this comment in related to https://github.com/mattermost/mattermost/pull/28789#discussion_r1816418394

RS-labhub replied:
Fixed!

hanzei replied:
Sorry, this is still unresolved.

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -102,8 +102,12 @@ func TestUpdateBookmark(t *testing.T) {
+		}()
 
 		bookmark2 := createBookmark("File to be updated", model.ChannelBookmarkFile, th.BasicChannel.Id, file.Id)
 		bookmarkResp, err := th.App.CreateChannelBookmark(th.Context, bookmark2, "")
```
We are now getting a conflict between `error` and `*model.AppError`. The clean solution would be to rename this `err` variable to `appErr`.

RS-labhub replied:
Done!

hanzei replied:
The issue isn't fixed yet. Do you need any help?

hanzei replied:
This is still a concern. See the CI error:

```

=== FAIL: channels/app TestUpdateBookmark/same_user_update_a_channel_bookmark (re-run 1) (0.01s)
    channel_bookmark_test.go:114: 
        	Error Trace:	/mattermost/server/channels/app/channel_bookmark_test.go:114
        	            				/mattermost/server/channels/app/channel_bookmark_test.go:168
        	Error:      	Received unexpected error:
        	            	<nil>
        	Test:       	TestUpdateBookmark/same_user_update_a_channel_bookmark
```

hanzei replied:
@RS-labhub Do you have any questions regarding the feedback?

RS-labhub replied:
@hanzei check this: https://github.com/mattermost/mattermost/pull/28789#issuecomment-2468009428

hanzei replied:
You are missing up the `error` type and the `*model.AppError` type. This line needs to become a new variable as `err` is of type `error`.

hanzei reviewed (changes requested):
We are getting there!

RS-labhub reviewed (commented):
Done!

rahimrahman commented:
@hanzei I'm going to remove myself as a reviewer.  My understanding of Go is very basic, I appreciate reading the code here.  But I don't feel comfortable giving a proper review and approval. Please assign another reviewer.  Thanks.

RS-labhub reviewed (commented):
Done

RS-labhub commented:
I renamed `err` to `appErr` but I'm not able to see my changes. That's weird. Let me get back to this branch and check it out. As far as I remember, I changed it!

RS-labhub commented:
> I renamed `err` to `appErr` but I'm not able to see my changes. That's weird. Let me get back to this branch and check it out. As far as I remember, I changed it!

Hey @hanzei I was right, I did the changes:
![image](https://github.com/user-attachments/assets/c6a13a96-47be-4d94-8417-acc6c5930efb)

I don't why it's not showing in my pushed commit 🙃

RS-labhub commented:
Found the error and fixed it!

RS-labhub commented:
Now it's fixed for sure!

hanzei reviewed (changes requested):
We are getting here

```

Error: channels/app/channel_bookmark_test.go:109:4: calling assert.Nil on error, please use assert.NoError instead
Error: channels/app/channel_bookmark_test.go:138:4: calling assert.Nil on error, please use assert.NoError instead
```

RS-labhub commented:
> We are getting here
> 
> ```
> 
> Error: channels/app/channel_bookmark_test.go:109:4: calling assert.Nil on error, please use assert.NoError instead
> Error: channels/app/channel_bookmark_test.go:138:4: calling assert.Nil on error, please use assert.NoError instead
> ```

If I do `require.NoError` then test case fails, if I do `require.Nil` then test case also fails. 

POV: It's an appErr, therefore it should be `require.Nil`

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -102,12 +102,16 @@ func TestUpdateBookmark(t *testing.T) {
+			assert.Nil(t, appErr)
+		}()
 
 		bookmark2 := createBookmark("File to be updated", model.ChannelBookmarkFile, th.BasicChannel.Id, file.Id)
 		bookmarkResp, err := th.App.CreateChannelBookmark(th.Context, bookmark2, "")
```
```suggestion
		bookmark2 := createBookmark("File to be updated", model.ChannelBookmarkFile, th.BasicChannel.Id, file.Id)
		bookmarkResp, appErr := th.App.CreateChannelBookmark(th.Context, bookmark2, "")
```

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -102,12 +102,16 @@ func TestUpdateBookmark(t *testing.T) {
 		bookmark2 := createBookmark("File to be updated", model.ChannelBookmarkFile, th.BasicChannel.Id, file.Id)
 		bookmarkResp, err := th.App.CreateChannelBookmark(th.Context, bookmark2, "")
-		require.Nil(t, err)
+		require.NoError(t, err)
```
```suggestion
		require.Nil(t, appErr)
```

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -102,12 +102,16 @@ func TestUpdateBookmark(t *testing.T) {
+		_, err := th.App.Srv().Store().FileInfo().Save(th.Context, file)
+		assert.NoError(t, err)
+		defer func() {
+			appErr := th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file.Id)
+			assert.Nil(t, appErr)
```
This is an an `error`
```suggestion
			err = th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file.Id)
			assert.NoError(t, err)
```

RS-labhub commented:
I hope it's all good now!

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -125,13 +129,18 @@ func TestUpdateBookmark(t *testing.T) {
+		err = th.App.Srv().Store().FileInfo().AttachToPost(th.Context, file2.Id, model.NewId(), th.BasicChannel.Id, model.BookmarkFileOwner)
+		assert.NoError(t, err)
+		defer func() {
+			appErr := th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file2.Id)
+			assert.Nil(t, appErr)
```
It is getting better! Now this also needs to become an `err` (and `require.NoError` should be used)

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -353,7 +369,10 @@ func TestUpdateChannelBookmarkSortOrder(t *testing.T) {
-	defer th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file.Id)
+	defer func() {
+		err = th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file.Id)
+		require.Nil(t, err)
```
```suggestion
		require.NoError(t, err)
```

hanzei reviewed (changes requested):
It is getting better! :tada:

RS-labhub commented:
I messed up everything!! I hope it's good now 😑

Extremely Sorry for messing this PR

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -125,13 +129,18 @@ func TestUpdateBookmark(t *testing.T) {
+		err = th.App.Srv().Store().FileInfo().AttachToPost(th.Context, file2.Id, model.NewId(), th.BasicChannel.Id, model.BookmarkFileOwner)
+		assert.NoError(t, err)
+		defer func() {
+			err := th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file2.Id)
```
Let's re-use the `err` variable
```suggestion
			err = th.App.Srv().Store().FileInfo().PermanentDelete(th.Context, file2.Id)
```

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -125,13 +129,18 @@ func TestUpdateBookmark(t *testing.T) {
+		}()
 
 		bookmark2.FileId = file2.Id
 		bookmarkResp, err = th.App.CreateChannelBookmark(th.Context, bookmark2, "")
```
```suggestion
		bookmarkResp, appErr := th.App.CreateChannelBookmark(th.Context, bookmark2, "")
```

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -125,13 +129,18 @@ func TestUpdateBookmark(t *testing.T) {
 		bookmark2.FileId = file2.Id
 		bookmarkResp, err = th.App.CreateChannelBookmark(th.Context, bookmark2, "")
-		require.NotNil(t, err)
+		require.Error(t, err)
```
```suggestion
		require.Nil(t, appErr)
```

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -125,13 +129,18 @@ func TestUpdateBookmark(t *testing.T) {
 		bookmark2.FileId = file2.Id
-		bookmarkResp, err = th.App.CreateChannelBookmark(th.Context, bookmark2, "")
-		require.NotNil(t, err)
+		bookmarkResp, appErr := th.App.CreateChannelBookmark(th.Context, bookmark2, "")
```
```suggestion
		bookmarkResp, appErr = th.App.CreateChannelBookmark(th.Context, bookmark2, "")
```

hanzei commented on server/channels/app/channel_bookmark_test.go:
```diff
@@ -125,13 +129,18 @@ func TestUpdateBookmark(t *testing.T) {
-		bookmarkResp, err = th.App.CreateChannelBookmark(th.Context, bookmark2, "")
-		require.NotNil(t, err)
+		bookmarkResp, appErr := th.App.CreateChannelBookmark(th.Context, bookmark2, "")
+		require.Nil(t, appErr)
```
```suggestion
		require.NotNil(t, appErr)
```

RS-labhub commented:
@hanzei @gabrieljackson all checks are passed now. Finally merging time!

hanzei reviewed (approved):
Finally :rocket:

gabrieljackson reviewed (approved):
LGTM
